### DISCUSSION TITLE
How to link issues and releases in a project table

### DISCUSSION THREAD

**Discussion by Mirko-von-Leipzig** [community/community#60935] [Projects and Issues]
### Select Topic Area

Product Feedback

### Feature Area

Projects

### Body

How can I automate a bug tracker which includes the release (if any) that fixes the bug?

I would like an extra column in my table view which indicates the release tag / name that includes the bug fix for an issue. How can I achieve this?

I can get "bug" issues by labelling them as such. In addition to the status column (todo, in-progress, done) I'd like to display which release the bug is resolved in (if any). 

There is the milestone column which can get added, but it requires manually creating the milestone and managing each issue to include it and isn't directly the equivalent of a release. Is there an easy way to automate this or to get a release-like column without doing it manually? 

I've managed to create a new milestone each time I create a release, however its been impossible to track which issues should be linked to the new milestone. Is there a somewhat reasonable way of achieving this?

I'd like to have a project view in which clients can track bugs and the release in which they are fixed (and not just when they're fixed in dev).

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to the GitHub feature areas [Issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-issues) and/or [Projects](https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects).

**Comment by lkiii** [comment]
I would like to see such capability as well. I want to see the board items by release and for that I would like to add the release to each of the issues/prs on the board

So far my ideas was quite manual, just a custom field with sting values of the release versions, but would be great to have more polished approach for this

**Comment by beenish-1** [comment]
I’d love to see this capability as well. The ideal workflow would let you **track which release fixes each bug directly on the project board**, without manually updating milestones or custom fields.

Currently, a few approaches can get close:

- **Custom field with release version**: You can create a custom field in your project (e.g., `Fixed In Release`) and manually enter the release tag. This works, but it can become tedious for many issues.

- **Automation via GitHub Actions**: When a release is published, a workflow could automatically update the `Fixed In Release` field or add a label to any issues or PRs merged in that release. This reduces manual work and keeps the board up to date.

- **Linking PRs to releases**: Encourage PRs to reference issues (`Closes #issue`) and include the release branch or tag in the PR description. Scripts or actions can then propagate that information to the issue or project board.

- **Third-party tools**: Some tools like ZenHub or Linear can track issues against releases more automatically and display them on boards.

Ideally, GitHub Projects would offer a **first-class “release column” or automatic release mapping**, so boards could show both bug status and the release it will appear in without any manual tracking.

**Reply by ebndev** [reply]
Hi @beenish-1, thanks for participating in GitHub Discussions!

Please note: we’ve clarified our stance on using generative AI tools like ChatGPT within our Community via [this announcement](https://github.com/orgs/community/discussions/69006). Please review the guidelines to ensure your post meets them as failure to adhere to those rules can result in action taken by our moderator team.  

You can read our updated [Code of Conduct](https://github.com/community/community/blob/baa5cd9020ef3f2bbf86dca1191d9c90f9b0822d/CODE_OF_CONDUCT.md#reasonable-use-of-ai-generated-content) and the [announcement](https://github.com/orgs/community/discussions/69006) for more details. 

Please feel encouraged to continue engaging in Discussions as your authentic self though. We appreciate you being a part of the Community 🌟

**Comment by CalderMatt09** [comment]
In practice, most teams approximate this workflow by combining milestones, pull request references, and automation. A common pattern is to align milestones with planned releases. When a pull request that fixes an issue is merged, the issue can automatically close if the PR references it. If that PR is included in a release branch or tag tied to the milestone, the milestone effectively becomes the indicator of the release in which the fix ships.

For additional visibility in a project table, many teams add a custom field such as “Release Version” or “Fixed In” and populate it through automation. This can be done with GitHub Actions or scripts that update the field when a PR associated with the issue is merged into a release branch. That way the project view can show status, bug label, and the release version in one place.

Workflows like this become even more valuable when stakeholders want to track defects from discovery through validation and release. Pairing issue tracking with structured QA workflows and [software test management tools](https://tuskr.app/) can help ensure bugs are not only fixed in code but also verified and documented before being associated with a specific release.
